# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositor.h
# Commit: f3b490c076d6
# Full Hash: f3b490c076d623d6ef3bcfd297a0522d9727477c
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-23 21:42:24
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 0c5fb3a8a94d -r f3b490c076d6 gfx/webrender_bindings/RenderCompositor.h
--- a/gfx/webrender_bindings/RenderCompositor.h	Thu Jan 23 15:35:45 2020 +0200
+++ b/gfx/webrender_bindings/RenderCompositor.h	Thu Jan 23 04:37:28 2020 +0000
@@ -79,6 +79,8 @@
 
   virtual bool IsContextLost();
 
+  virtual bool SupportAsyncScreenshot() { return true; }
+
   virtual bool ShouldUseNativeCompositor() { return false; }
   virtual uint32_t GetMaxUpdateRects() { return 0; }
 
@@ -95,6 +97,7 @@
   virtual void DestroyTile(wr::NativeSurfaceId, int32_t aX, int32_t aY) {}
   virtual void AddSurface(wr::NativeSurfaceId aId, wr::DeviceIntPoint aPosition,
                           wr::DeviceIntRect aClipRect) {}
+  virtual void EnableNativeCompositor(bool aEnable) {}
 
   // Interface for partial present
   virtual bool UsePartialPresent() { return false; }